### PR TITLE
Crosswalk Update - Partial Weams Match

### PR DESCRIPTION
**Pull request by devin-mccurdy**
## Description

- Added a new page for users to view a `CrosswalkIssue` along with related date and create or update a `Crosswalk` record to resolve the issue. This will eventually be used as a replacement for the Crosswalk upload.

If the updating `Crosswalk` fully resolves the issue (the `cross` and `ope` fields of the related `Weam`, `Crosswalk`, and `Ipeds_HD` records match) then the `CrosswalkIssue` is deleted.

- Updated `CrosswalkIssue.rebuild` to exclude Weams based on `poo_status` and `applicable_law_code` 

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/4751

https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/va.gov-team/5095

## Testing done
Tested locally and QA reviewed

## Screenshots

`crosswalk_issues/partials`
image

`crosswalk_issues/partials/#id`
image

## Acceptance criteria
- [x] Facilities in the WEAMS file that do not have a "POO Status" of "Approved" are excluded from the dashboard.

- [x] Facilities in the WEAMS file that have one of the following "Applicable Law Codes" are excluded from the dashboard:
  - blank
  - "educational institution is not approved"
  - "educational institution is approved for chapter 31 only"

- [x] The 'Crosswalk' column has an 'Action' sub-column.

- [x] The following actions are available for each row in the column:
  - a. Add blank OPE/IPEDS data to Crosswalk
  - b. Add WEAMS OPE/IPEDS data to Crosswalk
  - c. Add IPEDSHD OPE/IPEDS data to Crosswalk
  - d. Manually add OPE/IPEDS data to Crosswalk

- [x] If the user takes the "Add blank OPE/IPEDS data to Crosswalk" action; the crosswalk data source is updated with an entry that contains the following information:
  - a. facility code: matches the WEAMS data
  - b. institution name: matches the WEAMS data
  - c. city: matches the WEAMS data
  - d. state: matches the WEAMS data
  - e. ipeds: [blank]
  - f. ope: [blank]
  - g. notes: [user entered]

- [x] If the user takes the "Add WEAMS OPE/IPEDS data to Crosswalk" action; the crosswalk data source is updated with an entry that contains the following information:
  - a. facility code: matches the WEAMS data
  - b. institution name: matches the WEAMS data
  - c. city: matches the WEAMS data
  - d. state: matches the WEAMS data
  - e. ipeds: matches the WEAMS data
  - f. ope: matches the WEAMS data
  - g. notes: [user entered]

- [x] If the user takes the "Add IPEDSHD OPE/IPEDS data to Crosswalk" action; the crosswalk data source is updated with an entry that contains the following information:
  - a. facility code: matches the WEAMS data
  - b. institution name: matches the WEAMS data
  - c. city: matches the WEAMS data
  - d. state: matches the WEAMS data
  - e. ipeds: matches the IPEDS HD data
  - f. ope: matches the IPEDS HD data
  - g. notes: [user entered]

- [x] If the user takes the "Manually add OPE/IPEDS data to Crosswalk" action; the crosswalk data source is updated with an entry that contains the following information:
  - a. facility code: matches the WEAMS data
  - b. institution name: matches the WEAMS data
  - c. city: matches the WEAMS data
  - d. state: matches the WEAMS data
  - e. ipeds: matches the user specified data
  - f. ope: matches the user specified data
  - g. notes: [user entered]

- [x] The user is displayed a preview of the data that will be updated within the crosswalk.

- [x] On save:
  - a. If a crosswalk entry exists, the entry is updated with the data.
  - b. If a crosswalk entry does not exist, an entry is created with the data.
  - c. The row is removed from the "Crosswalk issues" table if the WEAMS IPEDS/OPE matches the Crosswalk IPEDS/OPE matches the IPEDS HD IPEDS/OPE values.

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by zurbergram — APPROVED** [review]

**Review by shawkey-daniel — APPROVED** [review]
LGTM

**Review by delli-gatti-michael — APPROVED** [review]

File: spec/controllers/crosswalk_issues_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+    login_user
+
+    before do
+      create_list :crosswalk_issue, 3, :partial_match_type
+      create_list :crosswalk_issue, 2, :ipeds_orphan_type
+      get(:orphans)
+    end
+
+    it 'populates an array of crosswalk issues' do
+      expect(assigns(:issues).length).to eq(CrosswalkIssue.by_issue_type(CrosswalkIssue::IPEDS_ORPHAN_TYPE).count)
```
Comment (johnpaulashenfelter) [inline spec/controllers/crosswalk_issues_controller_spec.rb]: Seems like you should be expecting the particular value here to make sure everything IS right. 0=0 is true, but might not be the numbers you're looking for if anything is off.

Comment (dmccurdybah) [reply spec/controllers/crosswalk_issues_controller_spec.rb]: Thanks!

File: spec/controllers/crosswalk_issues_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+        {
+          id: issue.id,
+          cross: '44445555',
+          ope: '333666',
+          notes: 'test'
+        }
+      end
+
+      it 'correctly updates Crosswalk fields from params' do
+        post(:resolve_partial, params: params)
```
Comment (jholton) [inline spec/controllers/crosswalk_issues_controller_spec.rb]: I know the context here is 'when related Crosswalk does not exist', but can we change this line to the following to make it clear that a `Crosswalk` object is created?
```suggestion
        expect{ post(:resolve_partial, params: params) }.to change(Crosswalk, :count).from(0).to(1)
```

File: spec/controllers/crosswalk_issues_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        expect(crosswalk[:cross]).to eq(params[:cross])
+        expect(crosswalk[:ope]).to eq(params[:ope])
+        expect(crosswalk[:notes]).to eq(params[:notes])
+      end
+
+      it 'correctly updates Crosswalk fields from weams' do
+        post(:resolve_partial, params: params)
```
Comment (jholton) [inline spec/controllers/crosswalk_issues_controller_spec.rb]: Same as above, can we change to
```suggestion
        expect{ post(:resolve_partial, params: params) }.to change(Crosswalk, :count).from(0).to(1)
```

File: spec/controllers/crosswalk_issues_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+        expect(crosswalk[:facility_code]).to eq(weam[:facility_code])
+        expect(crosswalk[:institution]).to eq(weam[:institution])
+        expect(crosswalk[:city]).to eq(weam[:city])
+        expect(crosswalk[:state]).to eq(weam[:state])
```
Comment (jholton) [inline spec/controllers/crosswalk_issues_controller_spec.rb]: `weam` city and state are `nil`.  Can you give those values in the factory, so it's clear that it is setting them on `crosswalk`?

**Review by jholton — APPROVED** [review]

**Pull request merged**